include/devicetree.h

```diff
@@ -2213,6 +2282,23 @@
 			      DT_DRV_COMPAT)(fn)),		\
 		    ())
 
+/**
+ * @brief Call "fn" on all nodes with compatible DT_DRV_COMPAT
+ *        and status "okay" with multiple arguments
+ *
+ *
+ * @param fn Macro to call for each enabled node. Must accept an
+ *           instance number as its only parameter.
+ * @param ... variable number of arguments to pass to fn
+ *
+ * @see DT_INST_FOREACH_STATUS_OKAY
+ */
+#define DT_INST_FOREACH_STATUS_OKAY_VARGS(fn, ...) \
+	COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),	\
+		    (UTIL_CAT(DT_FOREACH_OKAY_INST_VARGS_,	\
```

Comment:
Please use DT_CAT, not UTIL_CAT. We try not to use the util macros because their error messages are harder to read.

Reply:
I followed `DT_INST_FOREACH_STATUS_OKAY`. Should I update that as well?

Reply:
Sure, if you want to tack on another patch with that; thanks.